### PR TITLE
VersionRange raise ValueError if upper is not bigger than lower.

### LINKED ISSUE #37: If upper, lower and compared versions are same it returns false
https://github.com/canonical/charmed-openstack-upgrader/blob/a4101a6972c1191628d95d1b24e2246a632054b5/cou/utils/openstack.py#L244

For example:

```
arr = [VersionRange("6.0.0", "6.0.0")]
if "6.0.0" in arr:
    print("I am in range")
else:
    print("I am not in range")

```

### PR DESCRIPTION
**Pull request by gabrielcocenza**
Closes #37

### CODE REVIEW

File: cou/utils/openstack.py [hunk 1]
```diff
@@ -253,6 +253,11 @@ class VersionRange:
     lower: str
     upper: str
 
+    def __post_init__(self) -> None:
+        """Initialize the VersionRange dataclass and check values."""
+        if Version(self.lower) >= Version(self.upper):
+            raise ValueError("Upper version it is not bigger than lower version.")
```
Comment (agileshaw) [inline cou/utils/openstack.py:259]:
```suggestion
            raise ValueError("The upper bound version is not higher than the lower bound version.")
```

File: cou/utils/openstack.py [hunk 2]
```diff
@@ -253,6 +253,11 @@ class VersionRange:
     lower: str
     upper: str
 
+    def __post_init__(self) -> None:
+        """Initialize the VersionRange dataclass and check values."""
```
Comment (agileshaw) [inline cou/utils/openstack.py:257]:
```suggestion
        """Initialize the VersionRange dataclass and check its values."""
```

**Review by agileshaw — CHANGES_REQUESTED** [review]
Overall LGTM, just some wording changes suggested inline

File: cou/utils/openstack.py (hunk 1 above)
Comment (gabrielcocenza) [reply cou/utils/openstack.py]: Done

File: cou/utils/openstack.py (hunk 2 above)
Comment (gabrielcocenza) [reply cou/utils/openstack.py]: Done

**Review by agileshaw — APPROVED** [review]
+1 from me. Thanks :)

**Review by esunar — APPROVED** [review]

**Pull request merged**